tests/integration/files/test_files.py

```diff
@@ -43,37 +49,41 @@ def test_openai_client_basic_operations(compat_client, client_with_models):
 
         # Retrieve file content - OpenAI client returns httpx Response object
         content_response = client.files.content(uploaded_file.id)
-        # The response is an httpx Response object with .content attribute containing bytes
-        if isinstance(content_response, str):
-            # Llama Stack Client returns a str
-            # TODO: fix Llama Stack Client
-            content = bytes(content_response, "utf-8")
-        else:
-            content = content_response.content
-        assert content == test_content
+        assert content_response.content == test_content
 
         # Delete file
         delete_response = client.files.delete(uploaded_file.id)
         assert delete_response.deleted is True
 
-    except Exception as e:
-        # Cleanup in case of failure
-        try:
+        # Retrieve file should fail
+        with pytest.raises(NotFoundError, match="not found"):
+            client.files.retrieve(uploaded_file.id)
+
+        # File should not be found in listing
+        files_list = client.files.list()
+        file_ids = [f.id for f in files_list.data]
+        assert uploaded_file.id not in file_ids
+
+        # Double delete should fail
+        with pytest.raises(NotFoundError, match="not found"):
             client.files.delete(uploaded_file.id)
-        except Exception:
-            pass
-        raise e
 
+    finally:
+        # Cleanup in case of failure
+        if "uploaded_file" in locals():  # maybe create fails
+            try:
+                client.files.delete(uploaded_file.id)
+            except NotFoundError:
+                pass  # ignore 404
 
+
+@pytest.mark.xfail(message="Not all providers support user isolation")
```

Comment:
this is a really big deal. @ehhuang went through all the APIs and ensured that this worked. not having this work will punch a hole through the distribution guarantees now, won't it? if you want stuff deployed in an enterprise environment, you do want APIs to behave in a consistent way. 

why do we need to relax this?

Reply:
afaict, this test was always skipped.

i think it needs to be fixed, but i want to highlight it here and have it fixed later.

Reply:
I still think the message here needs to change. It cannot be that not all providers support user isolation but "user isolation is broken for current providers, must be fixed"

Reply:
done